keanu-project/src/main/java/io/improbable/keanu/tensor/dbl/DoubleTensor.java

```diff
@@ -70,6 +70,9 @@ static DoubleTensor placeHolder(int[] shape) {
     @Override
     DoubleTensor reshape(int... newShape);
 
+    @Override
+    DoubleTensor duplicate();
```

Comment:
Feels like this should be on `Tensor` - I imagine boolean and generic implementations will want it?

Reply:
It's on Tensor and GenericTensor but wasn't defined on the BooleanTensor interface. Added it.